src/main/java/io/ebeaninternal/server/transaction/DefaultTransactionThreadLocal.java

```diff
@@ -123,4 +118,13 @@ public static void end(String serverName) {
     }
   }
 
+  /**
+   * This is for testing purposes only. It will return TRUE, if a TransactionLocal has stuck.
+   */
+  public static boolean clear() {
```

Comment:
I think this is a question of whether DefaultTransactionThreadLocal can be garbage collected vs whether it is empty. That is, on tomcat webapp reload the expectation is that the threads are stopped and webapp classloader should be garbage collected (but the DefaultTransactionThreadLocal isn't being GC'ed at the moment and that I believe that leads to the tomcat webapp classloader not being GC'ed due to the reference to the TransactionMap class).